### PR TITLE
Update HTTP protocol to use ECS fields

### PR DESCRIPTION
**Pull request by andrewkroh**
NOTE: This is based on #9941 so ignore the first two commits. They will be removed after that PR merges.

Here's a summary of what fields changed.

Changed

- bytes_in -> source.bytes
- bytes_out -> destination.bytes
- http.request.body -> http.request.body.content
- http.response.body -> http.response.body.content
- http.response.code -> http.response.status_code
- http.response.phrase -> http.response.status_phrase
- method -> http.request.method (lowercased)
- notes -> error.message
- params -> url.query
- path -> url.path
- real_ip -> network.forwarded_ip
- responsetime -> event.duration (unit are now nanoseconds)
- transport -> network.transport

Added

- event.dataset = http
- event.end
- event.start
- http.request.referrer (always added if Referer header is present)
- http.version
- network.bytes
- network.community_id
- network.protocol = http
- network.type
- source.domain (added if Host header is present and not an IP address)
- url.domain - set with the Host header value
- url.full (synthesized from data in the request/response)
- url.port (when port is != 80)
- user_agent.original - (always added if User-Agent header is present)

Unchanged Packetbeat Fields

- query = {{ http.request.method }} {{ url.path }}
- request - text representation of the entire request
- response - text representation of the entire response
- status
- type = http (we might remove this since we have event.dataset)

### CODE REVIEW

File: packetbeat/protos/http/_meta/fields.yml [hunk 1]
```diff
@@ -10,10 +10,6 @@
           description: HTTP request
           type: group
           fields:
-            - name: params
```
Comment (ruflin) [inline packetbeat/protos/http/_meta/fields.yml]: Should this become an alias type with `migration: true`? Same for the other fields which were removed.

File: packetbeat/_meta/kibana/6/dashboard/Packetbeat-overview.json [hunk 2]
```diff
@@ -1,1164 +1,1182 @@
...
+        "title": "Response times percentiles",
+        "uiStateJSON": {},
+        "version": 1,
+        "visState": {
+          "aggs": [
+            {
+              "enabled": true,
+              "id": "1",
+              "params": {
+                "field": "responsetime",
```
Comment (adriansr) [inline packetbeat/_meta/kibana/6/dashboard/Packetbeat-overview.json]: shouldn't this be event.duration?

File: packetbeat/protos/http/_meta/fields.yml (hunk 1 above)
Comment (andrewkroh) [reply packetbeat/protos/http/_meta/fields.yml]: I thought this is what the ecs-migration's `alias: true` was going to be used for. So for any aliases that I want I should put them in fields.yml?

File: packetbeat/_meta/kibana/6/dashboard/Packetbeat-overview.json (hunk 2 above)
Comment (andrewkroh) [reply packetbeat/_meta/kibana/6/dashboard/Packetbeat-overview.json]: It should. Despite the huge diff I only changed the "Web Transactions" visualization of the overview dashboard. I will cycle back towards the end of this effort and do all of the general parts of the overview dashboard.

File: packetbeat/protos/http/_meta/fields.yml (hunk 1 above)
Comment (ruflin) [reply packetbeat/protos/http/_meta/fields.yml]: Yes, you need to put them in both places.

Comment (andrewkroh) [reply packetbeat/protos/http/_meta/fields.yml]: I added aliases for the HTTP fields where possible.

**Comment by andrewkroh** [comment]
Rebased to resolve an ecs-migration.yml conflict.

File: packetbeat/_meta/kibana/6/dashboard/Packetbeat-overview.json (hunk 2 above)
Comment (ruflin) [reply packetbeat/_meta/kibana/6/dashboard/Packetbeat-overview.json]: I would hope this PR does it automatically for you: https://github.com/elastic/beats/pull/9998

**Pull request merged**